covert/blockstream.py

```diff
@@ -33,17 +35,23 @@ def __init__(self):
     self.workers = 8
     self.executor = ThreadPoolExecutor(max_workers=self.workers)
     self.blkhash = None
+    self.file = None
     self.ciphertext = None
     self.q = collections.deque()
     self.pos = 0  # Current position within self.ciphertext; queued for decryption, not decoded
+    self.end = 0
+
 
   def authenticate(self, anykey):
     """Attempt decryption using secret key or password hash"""
-    if isinstance(anykey, pubkey.Key):
+    if isinstance(anykey, ratchet.Ratchet):
+      self.header.try_ratchet(anykey)
+    elif isinstance(anykey, pubkey.Key):
       self.header.try_key(anykey)
     else:
       self.header.try_pass(anykey)
 
+  @contextmanager
```

Comment:
For clarity, decrypt.py:DecryptView class should call decrypt_init() inside a with statement.

Reply:
There is no decrypt.py or DecryptView class, and decrypt_init() is called within two `with` statements in blockstream.py and cli.py. Also, this does not seem relevant to this PR.